src/bp/core/services/stats/index.ts

```diff
@@ -0,0 +1,215 @@
+    @inject(TYPES.LicensingService) private licenseService: LicensingService,
+    @inject(TYPES.WorkspaceService) private workspaceService: WorkspaceService,
+    @inject(TYPES.CMSService) private cmsService: CMSService,
+    @inject(TYPES.AuthService) private authService: AuthService,
+    @inject(TYPES.UserRepository) private userRepository: UserRepository,
+    @inject(TYPES.Database) private database: Database
+  ) {}
+
+  public start() {
+    // tslint:disable-next-line: no-floating-promises
+    this.run()
+    setInterval(this.run.bind(this), ms('6 hours'))
+  }
+
+  private async run() {
+    const lock = await this.jobService.acquireLock(LOCK_RESOURCE, ms('6 hours') - ms('1 minute'))
+    if (lock) {
+      debug('Acquired lock')
+      await this.sendStats()
+    }
+  }
+
+  private async sendStats() {
+    const stats = await this.getStats()
+    debug('Sending stats: ', JSON.stringify(stats))
+    try {
+      await axios.post('https://telemetry.botpress.io/ingest', stats)
+    } catch (err) {
+      // silently fail (only while the telemetry endpoint is under construction)
+    }
+  }
+
+  private async getStats() {
+    const config = await this.config.getBotpressConfig()
+
+    return {
+      schema: '1.0.0',
+      timestamp: new Date().toISOString(),
+      uuid: uuid.v4(),
+      bots: {
+        count: await this.getBotsCount()
+      },
+      workspaces: {
+        count: await this.getWorkspacesCount(),
+        pipelines: {
+          stages: {
+            count: await this.getPipelineStagesCount()
+          }
+        }
+      },
+      flows: {
+        count: await this.getFlowCount()
+      },
+      nlu: {
+        intents: {
+          count: await this.getIntentsCount()
+        },
+        entities: {
+          count: await this.getEntitiesCount()
+        }
+      },
+      qna: {
+        count: await this.getQnaCount()
+      },
+      hooks: {
+        global: {
+          count: await this.getGlobalHooksCount()
+        }
+      },
+      actions: {
+        global: {
+          count: await this.getGlobalActionsCount()
+        },
+        bot: {
+          count: await this.getBotActionsCount()
+        }
+      },
+      contentElements: {
+        count: await this.getContentElementsCount()
+      },
+      server: {
+        externalUrl: process.EXTERNAL_URL,
+        botpressVersion: process.BOTPRESS_VERSION,
+        fingerprint: await this.getServerFingerprint(),
+        clusterEnabled: yn(process.CLUSTER_ENABLED, { default: false }),
+        machineUUID: await machineUUID(),
+        nodesCount: await this.jobService.getNumberOfSubscribers(),
+        os: process.platform,
+        totalMemoryBytes: os.totalmem(),
+        uptime: Math.round(process.uptime()),
+        bpfsStorage: process.BPFS_STORAGE,
+        dbType: this.database.knex.isLite ? 'sqlite' : 'postgresql'
+      },
+      license: {
+        type: process.IS_PRO_ENABLED ? 'pro' : 'ce',
```

Comment:
would it make more sense using a boolean instead (eg: isProEnabled, like isProAvailable?)

Reply:
Using a string (`pro` or `ce`) leaves room for more license types in the future. Using a boolean value would only allow 2 different licenses.